alacritty_terminal/src/term/mod.rs

```diff
@@ -2429,6 +2733,277 @@ mod tests {
         assert_eq!(term.grid.cursor.point, Point::new(Line(4), Column(0)));
     }
 
+    #[test]
+    fn damage_public_usage() {
+        let size = SizeInfo::new(10.0, 10.0, 1.0, 1.0, 0.0, 0.0, false);
+        let mut term = Term::new(&Config::default(), size, ());
+        // Reset terminal for partial damage tests since it's initialized as fully damaged.
+        term.reset_damage();
+
+        // Test that we damage input form [`Term::input`].
+        let left = term.grid.cursor.point.column.0;
+        term.input('d');
+        term.input('a');
+        term.input('m');
+        term.input('a');
+        term.input('g');
+        term.input('e');
+        let right = term.grid.cursor.point.column.0;
+
+        let mut damaged_lines = match term.damage(None) {
+            TermDamage::Full => panic!("Expected partial damage, however got Full"),
+            TermDamage::Partial(damaged_lines) => damaged_lines,
+        };
+        assert_eq!(damaged_lines.next(), Some(LineDamageBounds { line: 0, left, right }));
+        assert_eq!(damaged_lines.next(), None);
+        term.reset_damage();
+
+        // Check that selection we've passed was properly damaged.
+        let line = 1;
+        let left = 0;
+        let right = term.columns() - 1;
+        let mut selection =
+            Selection::new(SelectionType::Block, Point::new(Line(line), Column(3)), Side::Left);
+        selection.update(Point::new(Line(line), Column(5)), Side::Left);
+        let selection_range = selection.to_range(&term);
+
+        let mut damaged_lines = match term.damage(selection_range) {
+            TermDamage::Full => panic!("Expected partial damage, however got Full"),
+            TermDamage::Partial(damaged_lines) => damaged_lines,
+        };
+        let line = line as usize;
+        // Skip information about cursor on zero line, since we just check for selection.
```

Comment:
I'm not sure what you mean by this. Is this suggestion accurate?

```suggestion
        // Skip cursor damage information, since we're just testing selection.
```